lib/galaxy/schema/schema.py

```diff
@@ -0,0 +1,359 @@
+        title="Type - ID",
+        description="The type and the encoded ID of this item. Used for caching.",
+        example="dataset-616e371b2cc6c62e",
+    )
+    type: str = Field(
+        ...,
+        title="Type",
+        description="The type of this item.",
+    )
+    create_time: datetime = Field(
+        ...,
+        title="Create Time",
+        description="The time and date this item was created.",
+    )
+    update_time: datetime = Field(
+        ...,
+        title="Update Time",
+        description="The last time and date this item was updated.",
+    )
+    url: AnyUrl = UrlField
+    tags: TagCollection
+
+
+class HDASummary(HistoryItemSummary):
+    dataset_id: EncodedDatabaseIdField = Field(
+        ...,
+        title="Dataset ID",
+        description="The encoded ID of the dataset associated with this item.",
+    )
+    state: Dataset.states = DatasetStateField
+    extension: str = Field(
+        ...,
+        title="Extension",
+        description="The extension of the dataset.",
+        example="txt",
+    )
+    purged: bool = Field(
+        ...,
+        title="Purged",
+        description="Whether this dataset has been removed from disk.",
+    )
+
+
+class HDAInaccessible(HistoryItemBase):
+    accessible: bool = AccessibleField
+    state: Dataset.states = DatasetStateField
+
+
+class HDADetailed(HDASummary):
+    model_class: str = ModelClassField(HDA_MODEL_CLASS_NAME)
+    hda_ldda: str = Field(
+        "hda",
+        const=True,
+        title="HDA or LDDA",
+        description="Whether this dataset belongs to a history (HDA) or a library (LDDA).",
+        deprecated=False  # TODO Should this field be deprecated in favor of model_class?
+    )
+    accessible: bool = AccessibleField
+    genome_build: str = Field(
+        "?",
+        title="Genome Build",
+        description="TODO",
+    )
+    misc_info: str = Field(
+        ...,
+        title="Miscellaneous Information",
+        description="TODO",
+    )
+    misc_blurb: str = Field(
+        ...,
+        title="Miscellaneous Blurb",
+        description="TODO",
+    )
+    file_ext: str = Field(  # Is this a duplicate of HDASummary.extension?
+        ...,
+        title="File extension",
+        description="The extension of the file.",
+    )
+    file_size: int = Field(
+        ...,
+        title="File Size",
+        description="The file size in bytes.",
+    )
+    resubmitted: bool = Field(
+        ...,
+        title="Resubmitted",
+        description="Whether the job creating this dataset has been resubmitted.",
+    )
+    metadata: Any = Field(  # TODO: create pydantic model for metadata?
+        ...,
+        title="Metadata",
+        description="The metadata associated with this dataset.",
+    )
+    meta_files: List[MetadataFile] = Field(
+        [],
+        title="Metadata Files",
+        description="Collection of metadata files associated with this dataset.",
+    )
+    data_type: str = Field(
```

Comment:
This is kind of a duplicate of ext ... ext maps to a data_type. Maybe we should deprecate this at some later point ?
Returning the actual class is also not very stable, this is probably something that should have never left the API